deps/ada.cpp

```diff
@@ -15009,6 +15009,14 @@ uint8_t ada_get_host_type(ada_url result) noexcept {
   return r->host_type;
 }
 
+uint8_t ada_get_schema_type(ada_url result) noexcept {
+  ada::result<ada::url_aggregator>& r = get_instance(result);
+  if (!r) {
+    return 0;
+  }
+  return r->type;
```

Comment:
Can you open a separate pull request to ada repository for the c++ changes?

Reply:
Oh so the C++ is pulled in actually?

Reply:
I have a branch ready but can't push yet because I get a permission denied error. We then also have to update the C++ code here. How do you do that?

Reply:
You can fork to a repo you own then push there, next: issue a PR.

Reply:
Yea the PR is up here https://github.com/ada-url/ada/pull/503. Is that what you mean?

Reply:
Oh I see, so the dependency is just copied once there's a new version of ada.

Reply:
We need to run `./.github/update-ada.sh` command and open a new pull request once the new version is released.

Reply:
I ll submit a PR to update to the new version.